packages/api/tests/test-es-indexer.js

```diff
@@ -0,0 +1,552 @@
+'use strict';
+
+const test = require('ava');
+const sinon = require('sinon');
+const fs = require('fs');
+const path = require('path');
+const aws = require('@cumulus/common/aws');
+const { StepFunction } = require('@cumulus/ingest/aws');
+const { randomString } = require('@cumulus/common/test-utils');
+const indexer = require('../es/indexer');
+const { Search } = require('../es/search');
+const { bootstrapElasticSearch } = require('../lambdas/bootstrap');
+const granuleSuccess = require('./data/granule_success.json');
+const granuleFailure = require('./data/granule_failed.json');
+const pdrFailure = require('./data/pdr_failure.json');
+const pdrSuccess = require('./data/pdr_success.json');
+
+const esIndex = randomString();
+process.env.bucket = randomString();
+process.env.stackName = 'my-stack';
+process.env.ES_INDEX = esIndex;
+let esClient;
+
+
+test.before(async () => {
+  // create the elasticsearch index and add mapping
+  await bootstrapElasticSearch('fakehost', esIndex);
+  esClient = await Search.es();
+
+  // create buckets
+  await aws.s3().createBucket({ Bucket: process.env.bucket }).promise();
+});
+
+test.after.always(async () => {
+  // remove elasticsearch index
+  await esClient.indices.delete({ index: esIndex });
+  await aws.recursivelyDeleteS3Bucket(process.env.bucket);
+});
+
+test('indexing a successful granule record', async (t) => {
+  const type = 'granule';
+  const granule = granuleSuccess.payload.granules[0];
+  const collection = granuleSuccess.meta.collection;
+  const r = await indexer.granule(esClient, granuleSuccess, esIndex, type);
+
+  // make sure record is created
+  t.is(r[0].result, 'created');
+
+  const collectionId = indexer.constructCollectionId(collection.name, collection.version);
+
+  // check the record exists
+  const record = await esClient.get({
+    index: esIndex,
+    type,
+    id: granule.granuleId,
+    parent: collectionId
+  });
+
+  t.deepEqual(record._source.files, granule.files);
+  t.is(record._source.status, 'completed');
+  t.is(record._parent, collectionId);
+  t.is(record._id, granule.granuleId);
+  t.is(record._source.cmrLink, granule.cmrLink);
+  t.is(record._source.published, granule.published);
+
+  const { name: deconstructed } = indexer.deconstructCollectionId(record._parent);
+  t.is(deconstructed, collection.name);
+});
+
+test('indexing multiple successful granule records', async (t) => {
+  const newPayload = JSON.parse(JSON.stringify(granuleSuccess));
+  const type = 'granule';
+  const granule = newPayload.payload.granules[0];
+  granule.granuleId = randomString();
+  const granule2 = Object.assign({}, granule);
```

Comment:
This is going to perform a shallow clone.  Is that acceptable?

Reply:
yeah, here I want same granules info with different names only. This is on purpose.